calcom/cal.com
# feat: global booking limits for teams

## What does this PR do?

Adds a new feature for global booking limits for teams. Team global booking limits determine how many time team members can be booked from team event types (collective, round-robin)

![Screenshot 2024-09-17 at 6 14 02 PM](https://github.com/user-attachments/assets/c207102a-78eb-438e-b010-4e14fc1daa03)

- Fixes #16557
- Fixes CAL-4295

Also fixes an existing bug with booking limits when host and attendees were in different timezones. Slots didn't load correctly and when trying to confirm the booking it was throwing `Could not book the meeting.
Booking Limit for this event type has been reached`

- Fixes #14436
- Fixes CAL-3458
- Fixes #15005
- Fixes CAL-3692

Also fixes issues with booking limits (duration and frequency) when event is rescheduled. It sometimes caused slots not to show up or for duration limits slots were correctly shown but it was throwing `Could not reschedule the meeting. Duration Limit for this event type has been reached` after confirming the bookings
- Fixes #14584
- Fixes CAL-3413

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have added a Docs issue: https://github.com/calcom/docs/issues/139

## How should this be tested?

- Go to `settings/team/[id]/bookingLimits `
- Add some limits 
- book team event types (collective, round-robin, managed) and see if it works as expected
